src/manifest.c

```diff
@@ -897,7 +959,7 @@ void link_submanifests(struct manifest *m1, struct manifest *m2, struct list *su
 
 /* if component is specified explicitly, pull in submanifest only for that
  * if component is not specified, pull in any tracked component submanifest */
-struct list *recurse_manifest(struct manifest *manifest, struct list *subs, const char *component, bool server, int *err)
+struct list *recurse_manifest(int current_version, struct manifest *manifest, struct list *subs, const char *component, bool server, int *err)
```

Comment:
This requires that delta manifests for every bundle for 'N' previous versions are generated for every version, even when only one bundle was updated. This is currently done in clearlinux/mixer-tools#543, but I think it would be better to reference previous versions to get manifest deltas for bundles that were not changed in the 'to' version which is how the old Swupd implementation of manifest deltas worked.

Reply:
Agreed, I've updated the mixer-tools code to align with that. Sorry for the churn.

Reply:
Nice. Point. I'll fix that